onnxscript/evaluator.py

```diff
@@ -262,7 +274,11 @@ def eval_function(
         # Split happens in the evaluator instead of the OnnxFunction __call__ method
         # so that evaluators can control behaviors like whether to fill in default values for attributes.
         inputs, attributes = param_manipulation.separate_input_attributes_from_arguments(
-            param_schemas, args, kwargs, fill_defaults=False, allow_extra_kwargs=False
+            param_schemas,
+            args,
+            kwargs,
+            fill_defaults=False,
+            allow_extra_kwargs=self._ignore_unknown_function_kwargs,
```

Comment:
Not so sure how this works:

https://github.com/microsoft/onnx-script/blob/24b153ae2083527ad928c9be54d0b6b8dfb129a1/onnxscript/_internal/param_manipulation.py#L41-L42

Shouldn't it just raise an error if the differences are detected?

Reply:
So if allow_extra_kwargs and there is extra args then we don't raise; if allow_extra_kwargs is false and there is extra args then we raise. Was I missing something?

Reply:
(1) Are the extra kwargs silently ignored when no error is raised?
(2) What is the motivating use-case?

Reply:
I was looking for filter, but this seems just allowing it.

Reply:
(1) Yes
(2) One use case is in the TorchScriptEvaluator, where there is an unknown and changing number of kwargs in the torch ops, e.g. `memory_format` etc. that does not change the op's numerical behavior and thus not applicable in onnx. We initially use a block like to filter out those kwargs, but as torch and introduce new kwargs, I think it is safer for us to just ignore them so atenlib doesn't break as torch introduces these kwargs.